uniffi_bindgen/src/bindings/kotlin/templates/RustBufferHelpers.kt

```diff
@@ -13,6 +13,28 @@ fun<T> liftFromRustBuffer(rbuf: RustBuffer.ByValue, readItem: (ByteBuffer) -> T)
     }
 }
 
+fun<T> lowerIntoRustBuffer(v: T, writeItem: (T, RustBufferBuilder) -> Unit): RustBuffer.ByValue {
+    // TODO: maybe we can calculate some sort of initial size hint?
+    val buf = RustBufferBuilder()
+    try {
+        writeItem(v, buf)
+        return buf.finalize()
+    } catch (e: Throwable) {
+        buf.discard()
+        throw e
+    }
+}
+
+// For every type used in the interface, we provide helper methods for conveniently
+// lifting and lowering that type from C-compatible data, and for reading and writing
+// values of that type in a buffer.
+
+{% for typ in ci.iter_types() %}
+{% let type_name = typ.canonical_name()|class_name_kt %}
```

Comment:
Nit: naming. I've scrolled up to here to find out what this was. 

Would prefer  something more descriptive: `canonical_type_name` or `type_identifier`.